Framework/DataHandling/src/LoadSwans.cpp

```diff
@@ -0,0 +1,230 @@
+  loadInstrument();
+  m_detector_size = setDetectorSize();
+  std::map<uint32_t, std::vector<uint32_t>> pos_tof_map = LoadSwans::loadData();
+  loadDataIntoTheWorkspace(pos_tof_map);
+  loadInstrument();
+  setTimeAxis();
+  placeDetectorInSpace();
+  setProperty("OutputWorkspace", m_ws);
+}
+
+/**
+ * Run the Child Algorithm LoadInstrument.
+ * It sets the workspace with the necessary information
+ */
+void LoadSwans::loadInstrument() {
+
+  IAlgorithm_sptr loadInst = createChildAlgorithm("LoadInstrument");
+
+  // Now execute the Child Algorithm. Catch and log any error, but don't stop.
+  try {
+    loadInst->setPropertyValue("InstrumentName", m_instrumentName);
+    loadInst->setProperty<MatrixWorkspace_sptr>("Workspace", m_ws);
+    loadInst->setProperty("RewriteSpectraMap",
+                          Mantid::Kernel::OptionalBool(true));
+    loadInst->execute();
+  } catch (...) {
+    g_log.information("Cannot load the instrument definition.");
+  }
+}
+
+/**
+ * Place the detector in space according to the distance and angle
+ * Needs in the IDF Parameters file the entries:
+ * detector-name, detector-sample-distance, detector-rotation-angle
+ */
+void LoadSwans::placeDetectorInSpace() {
+
+  std::string componentName =
+      m_ws->getInstrument()->getStringParameter("detector-name")[0];
+  const double distance = static_cast<double>(
+      m_ws->getInstrument()->getNumberParameter("detector-sample-distance")[0]);
+  const double angle = static_cast<double>(
+      m_ws->getInstrument()->getNumberParameter("detector-rotation-angle")[0]);
+
+  g_log.information() << "Moving detector " << componentName << " " << distance
+                      << " meters and " << angle << " degrees." << std::endl;
+
+  LoadHelper helper;
+
+  const double deg2rad = M_PI / 180.0;
+  V3D pos = helper.getComponentPosition(m_ws, componentName);
+  double angle_rad = angle * deg2rad;
+  V3D newpos(distance * sin(angle_rad), pos.Y(), distance * cos(angle_rad));
+  helper.moveComponent(m_ws, componentName, newpos);
+
+  // Apply a local rotation to stay perpendicular to the beam
+  const V3D axis(0.0, 1.0, 0.0);
+  Quat rotation(angle, axis);
+  helper.rotateComponent(m_ws, componentName, rotation);
+}
+
+/**
+ * Load the data into a map. The map is indexed by pixel id (0 to 128*128-1 =
+ * m_detector_size)
+ * The map values are the events TOF
+ * @returns the map of events indexed by pixel index
+ */
+std::map<uint32_t, std::vector<uint32_t>> LoadSwans::loadData() {
+
+  std::string filename = getPropertyValue("Filename");
+  std::ifstream input(filename, std::ifstream::binary | std::ios::ate);
+  input.seekg(0);
+
+  m_ws->initialize(m_detector_size, 1, 1);
+
+  std::map<uint32_t, std::vector<uint32_t>> pos_tof_map;
+
+  while (input.is_open()) {
+    if (input.eof())
+      break;
+    uint32_t tof = 0;
+    input.read((char *)&tof, sizeof(tof));
+    tof -= static_cast<uint32_t>(1e9);
+    tof = static_cast<uint32_t>(tof * 0.1);
```

Comment:
Integer division would be let you avoid the static_cast. If you depend on rounding, use `std::round` before casting.